### PR TITLE
Improve error msg for non-reactive properties

### LINKED ISSUE #6657: improve error message for non-reactive properties
### What problem does this feature solve?
current error message is non helpful for vue-class-component (js or ts):

if you forget to initialize a property then you get this error:

`Property or method "xxx" is not defined on the instance but referenced during render. Make sure to declare reactive data properties in the data option.`

So of course with class based components there is no 'data option', so this isn't helpful.

Proposal is to change the error message:

`Property or method "xxx" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property.  See:  https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties`

### What does the proposed API look like?
no changes.

### PR DESCRIPTION
**Pull request by kutyel**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [x] Other, please describe:

Improved error message.

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**

Fixes #6657

### CODE REVIEW

File: src/core/instance/proxy.js [hunk 1]
```diff
@@ -42,7 +44,7 @@ if (process.env.NODE_ENV !== 'production') {
   }
 
   const hasHandler = {
-    has (target, key) {
```
Comment (yyx990803) [inline src/core/instance/proxy.js]: Can we remove the code style changes?

File: src/core/instance/proxy.js [hunk 2]
```diff
@@ -15,9 +15,11 @@ if (process.env.NODE_ENV !== 'production') {
...
     warn(
-      `Property or method "${key}" is not defined on the instance but ` +
-      `referenced during render. Make sure to declare reactive data ` +
-      `properties in the data option.`,
+      `Property or method "${key}" is not defined on the instance but
+      referenced during render. Make sure that this property is reactive,
+      either in the data option, or for class-based components, by
+      initializing the property.
+      See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.`,
       target
```
Comment (yyx990803) [inline src/core/instance/proxy.js]: We need to use concatenations for each line to avoid the whitespace between the lines.

**Comment by kutyel** [comment]
Thanks, @yyx990803 should be ok right now 👌

**Comment by posva** [comment]
@kutyel you can run `npm run lint -- --fix` to fix the lint problems

**Pull request merged**